src/demo-app/select/select-demo.html

```diff
@@ -113,5 +113,20 @@
     </md-card>
   </div>
 
+  <md-card *ngIf="showSelect">
+    <md-select placeholder="Drink" [(ngModel)]="currentDrink" #selectWitHeader="mdSelect">
+      <md-select-header>
+        <input type="search" [(ngModel)]="searchTerm" role="combobox" [attr.aria-owns]="selectWitHeader.panelId"
```

Comment:
Hmm you're right. This complicates things a lot, because it means that we'd need two different key handling strategies depending on whether the select has a header (`FocusKeyManager` by default and `ActiveDescendantKeyManager` if there's a header), which in turn will complicate the select even more, because it would have to delegate to the header in some cases but not in others. I can see how it can turn into a huge mess pretty quickly since we'd need to sprinkle `if (this.header)` all over the place and there would be a `MdSelect <-> MdSelectHeader` dependency.